addons/html_editor/static/src/main/table/table_plugin.js

```diff
@@ -305,6 +319,37 @@ export class TablePlugin extends Plugin {
             }
         }
     }
+    /**
+     * @param {HTMLTableRowElement} reference
+     */
+    makeHeader(reference) {
+        [...reference.children].forEach((td) => {
+            if (td.nodeName == "TD") {
+                const th = document.createElement("th");
```

Comment:
```suggestion
                const th = this.document.createElement("th");
```
same for `removeHeader`.